src/main/java/htsjdk/samtools/reference/AbstractIndexedFastaSequenceFile.java

```diff
@@ -146,6 +147,41 @@ protected static void sanityCheckDictionaryAgainstIndex(final String fastaFile,
         }
     }
 
+    /** Do some basic checking to make sure the fasta and the index match.
+     * <p>
+     * checks that the length of the fasta file is at least as long as the index proclaims
+     * and that beyond the last position references in the index there is only one line followed by whitespaces
+     *
+     * @param fastaFile Used for error reporting only.
+     * @param index index file to check against the dictionary.
+     */
+    public static void sanityCheckFastaAgainstIndex(final String fastaFile,
+                                                    final FastaSequenceIndex FastaSequenceIndex) {
+
+
+        final Iterator<FastaSequenceIndexEntry> iterator = FastaSequenceIndex.iterator();
+        FastaSequenceIndexEntry fastaSequenceIndex = null;
+        while (iterator.hasNext()) {
+            fastaSequenceIndex = iterator.next();
+        }
```

Comment:
It kinda sucks to have to do this iteration.  I wonder if it would make sense to expand the PR a little bit, and modify `FastaSequenceIndex` to either store index entries in a pair of (Array/ArrayList, HashMap (instead of LinkedHashMap))?  Or just to hold an extra pointer to the final entry in the index?

Reply:
There are several operations that explicitly take advantage of the linked-list aspect...come to think of it, I my sanity-check only works on a "freshly make" object, since operations such as "rename" remove and add entries, which moved the renamed entry to the "end" of the linked-list.... I only _need_ it to work upon initialization, but I need to add some protections. upshot is that a list would be overkill, I'll go with the reference to the last element.